derive/src/trait_bounds.rs

```diff
@@ -149,29 +160,49 @@ pub fn add(
 			.for_each(|ty| {
 				where_clause.predicates.push(parse_quote!(#ty : #has_compact_bound))
 			});
+
+		skip_types
+			.into_iter()
+			.for_each(|ty| {
+				let codec_skip_bound = codec_skip_bound.as_ref().unwrap();
+				where_clause.predicates.push(parse_quote!(#ty : #codec_skip_bound))
+			});
 	}
 
 	Ok(())
 }
 
-fn needs_codec_bound(field: &syn::Field) -> bool {
-	!crate::utils::get_enable_compact(field)
+fn needs_codec_bound(field: &syn::Field, variant_skip: bool) -> bool {
+	!variant_skip
+		&& !crate::utils::get_enable_compact(field)
 		&& crate::utils::get_encoded_as_type(field).is_none()
+		&& crate::utils::get_skip(&field.attrs).is_none()
 }
 
-fn needs_has_compact_bound(field: &syn::Field) -> bool {
-	crate::utils::get_enable_compact(field)
+fn needs_has_compact_bound(field: &syn::Field, variant_skip: bool) -> bool {
+	!variant_skip
+		&& crate::utils::get_enable_compact(field)
 }
 
-fn collect_types(data: &syn::Data, type_filter: fn(&syn::Field) -> bool) -> syn::Result<Vec<syn::Type>> {
+fn needs_default_bound(field: &syn::Field, variant_skip: bool) -> bool {
+	// types from skipped variants doesn't need default
+	!variant_skip
+		&& crate::utils::get_skip(&field.attrs).is_some()
+}
+
+fn collect_types(
+	data: &syn::Data,
+	// args: field, if inside a variant with skip attributes
+	type_filter: fn(&syn::Field, bool) -> bool,
+) -> syn::Result<Vec<syn::Type>> {
 	use syn::*;
 
 	let types = match *data {
 		Data::Struct(ref data) => match &data.fields {
 			| Fields::Named(FieldsNamed { named: fields , .. })
 			| Fields::Unnamed(FieldsUnnamed { unnamed: fields, .. }) => {
 				fields.iter()
-					.filter(|f| type_filter(f))
+					.filter(|f| type_filter(f, false))
```

Comment:
Why don't we check for `skip` here?

Reply:
we should not filter out skip fields here because collect_types is used to get skip fields to bound Default on them in Decode macro